### PR TITLE
Doc/redefine header

### PR DESCRIPTION
**Pull request by matthiasbeyer**
As proposed in personal discussion with @TheNeikos

### CODE REVIEW

File: doc/src/02000-store.md [hunk 1]
```diff
@@ -16,21 +16,41 @@ The content follows the header-close marker (@sec:thestore:fileformat:content).
...
+
+The "imag" section MAY contain
+
+1. A section "imag.links" where a module is allowed to store URIs in a flat
+   list
+1. A section "imag.content", used for refering to external content.
+   Valid keys in this section are "uri", "file" and "mime", whereas
+     - "uri" refers to the external content
+     - "file" refers to a local variant, this is optional
+     - "mime" stores the MIME-Type of the local variant, if present.
```
Comment (neithernut) [inline doc/src/02000-store.md]: This is redundant, as you can simply use the `file://` scheme to refer to local files.

File: doc/src/02000-store.md [hunk 2]
```diff
@@ -16,21 +16,41 @@ The content follows the header-close marker (@sec:thestore:fileformat:content).
...
+
+#### Header section: "module" {#sec:thestore:fileformat:header:module}
+
+The header MAY contain a section named after a module.
+The corrosponding module is allowed to store arbitrary data in this section.
```
Comment (neithernut) [inline doc/src/02000-store.md]: The title of this section is misleading as it suggests that a TOML-section _named_ `mudule` does exist. I propose renaming this section to something like `Module sections` or `Header section: module sections`.

Comment (matthiasbeyer) [inline doc/src/02000-store.md]: Thanks. I searched for a better naming but failed. Will name appropriately and move below the "custom" section.

**Comment by matthiasbeyer** [comment]
Ready for cleanup.

**Comment by neithernut** [comment]
`file` and `uri` are still redundant.

**Comment by matthiasbeyer** [comment]
No. If you reread the section, `uri` is for pointing _anywhere_ but `file` may point to a local variant.

I'm not that sure whether we want that. @TheNeikos what do you think?

**Comment by neithernut** [comment]
In this case `refers to the external content` is ambogious.

**Comment by neithernut** [comment]
Also: omit the `the`

**Comment by matthiasbeyer** [comment]
Hm, no. I can see now what you mean. Maybe @TheNeikos can point out what he meant by `file` ...

File: doc/src/02000-store.md [hunk 3]
```diff
@@ -16,21 +16,40 @@ The content follows the header-close marker (@sec:thestore:fileformat:content).
...
+
+The header MUST contain a section called "imag", where the automatically by the
+program generated data is stored in.
+The contents of this section is edited via commandline calls or by the
+program implicitely and SHOULD NOT be edited by the user.
+
+This "imag" section MUST contain the following keys
+
+1. A "version" Key. The version stored here is the version of the Store, the
+   Entry is compatible with.
```
Comment (TheNeikos) [inline doc/src/02000-store.md]: I would say that it was created with this version of the store, compability is not necessarily encoded in the version.

**Comment by matthiasbeyer** [comment]
@TheNeikos What about the `file` member, @neithernut and I argued over above?

**Comment by TheNeikos** [comment]
The idea was that you have an `url` field that is required, and then the module can put in w/e it wants there. Then you would put in optional fields with well-defined meanings to make sure libraries understand each other, like for example `file` and `mime` meant for example for a local representation of the entry.

**Comment by matthiasbeyer** [comment]
@TheNeikos as suggested...

File: doc/src/02000-store.md [hunk 4]
```diff
@@ -16,21 +16,40 @@ The content follows the header-close marker (@sec:thestore:fileformat:content).
...
+
+1. A "version" Key. The version stored here is the version of the Store, the
+   Entry was created with.
+
+The "imag" section MAY contain
+
+1. A section "imag.links" where a module is allowed to store URIs in a flat
+   list
+1. A section "imag.content", used for refering to external content.
+   Valid keys in this section are "uri" and "file", whereas
```
Comment (TheNeikos) [inline doc/src/02000-store.md]: only file is required, *valid* is everything needed by the library that it needs.

**Comment by neithernut** [comment]
Would a URI pointing to something local still be valid?

**Comment by TheNeikos** [comment]
Yup

**Comment by neithernut** [comment]
In this case the semantics kinda make sense. Maybe I'll come back with some clarifications at some point. But for now I can ACK this.

**Comment by matthiasbeyer** [comment]
@TheNeikos do you ACK as well?

**Comment by TheNeikos** [comment]
Yes

**Comment by matthiasbeyer** [comment]
@homu r+

**Pull request merged**